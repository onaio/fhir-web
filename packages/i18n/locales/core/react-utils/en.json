{
  "404": "",
  "Actions": "",
  "Class": "",
  "Date created": "",
  "Error": "",
  "Fetching next page": "",
  "Go back": "",
  "Go home": "",
  "ID": "",
  "Load more options": "",
  "No data": "",
  "No data available": "",
  "Search": "",
  "Showing {{recordsFetchedNum}}; {{remainingRecords}} more records.": "",
  "Something went wrong": "",
  "Sorry, the resource you requested for, does not exist": "",
  "Sorry, you are not authorized to access this page": "",
  "status": "",
  "Unable to load dropdown options.": "",
  "View details": ""
}
