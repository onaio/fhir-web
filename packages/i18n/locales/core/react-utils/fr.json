{
  "403": "",
  "404": "",
  "Actions": "",
  "Error": "",
  "Go back": "",
  "Go home": "",
  "Search": "",
  "Something went wrong": "",
  "Sorry, the resource you requested for, does not exist": "",
  "Sorry, you are not authorized to access this page": ""
}
