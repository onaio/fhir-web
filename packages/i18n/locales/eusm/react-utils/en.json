{
  "403": "",
  "404": "",
  "Actions": "",
  "Error": "",
  "Fetching next page": "Fetching next page",
  "Go back": "",
  "Go home": "",
  "Load more options": "Load more options",
  "No data": "No data",
  "Search": "",
  "Showing {{recordsFetchedNum}}; {{remainingRecords}} more records.": "Showing {{recordsFetchedNum}}; {{remainingRecords}} more records.",
  "Something went wrong": "",
  "Sorry, the resource you requested for, does not exist": "",
  "Sorry, you are not authorized to access this page": "",
  "Unable to load dropdown options.": "Unable to load dropdown options."
}
