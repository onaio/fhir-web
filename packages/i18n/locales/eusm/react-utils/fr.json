{
  "403": "403",
  "404": "404",
  "Actions": "Actions",
  "Error": "Erreur",
  "Go back": "Retour",
  "Go home": "Retour à l'accueil",
  "Search": "Recherche",
  "Session Expired": "Session expirée",
  "Something went wrong": "Il y a eu un problème",
  "Sorry, the resource you requested for, does not exist": "Désolé, la ressource que vous avez demandée n'existe pas",
  "Sorry, you are not authorized to access this page": "Désolé, vous n'êtes pas autorisé à accéder à cette page"
}
