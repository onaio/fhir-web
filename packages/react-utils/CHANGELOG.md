# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.5](https://github.com/opensrp/web/compare/@opensrp/react-utils@0.0.4...@opensrp/react-utils@0.0.5) (2021-02-23)

**Note:** Version bump only for package @opensrp/react-utils

## [0.0.4](https://github.com/opensrp/web/compare/@opensrp/react-utils@0.0.3...@opensrp/react-utils@0.0.4) (2021-02-18)

**Note:** Version bump only for package @opensrp/react-utils

## 0.0.3 (2021-02-18)

**Note:** Version bump only for package @opensrp/react-utils

## 0.0.1 (2020-11-12)

**Note:** Version bump only for package @opensrp/react-utils
