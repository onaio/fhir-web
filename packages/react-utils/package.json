{
  "name": "@opensrp/react-utils",
  "version": "0.0.6",
  "description": "A couple of utils that work with react",
  "main": "dist/index.js",
  "types": "dist/types",
  "files": [
    "dist",
    "locales"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/opensrp/web",
  "config": {
    "folderName": "react-utils"
  },
  "scripts": {
    "test": " cd ../../ && yarn test packages/$npm_package_config_folderName --verbose --collectCoverage=true && cd packages/$npm_package_config_folderName",
    "tsc": "tsc",
    "lint": "eslint './**/*.{js,jsx,ts,tsx}'",
    "copy": "copyfiles -u 1 \"./src/**/*.{css,html}\" \"./dist/\"",
    "build": "yarn run tsc && yarn run transpile && yarn run copy",
    "transpile": "babel src -d dist --root-mode upward --extensions '.ts,.tsx' --ignore '**/*.test.ts,**/*.test.tsx,**/tests,**/__tests__'",
    "transpile:windows": "babel src -d dist --root-mode upward --extensions .ts,.tsx --ignore '**/*.test.ts,**/*.test.tsx,**/tests,**/__tests__'",
    "extractTranslations": "yarn i18next 'src/**/*.{ts,tsx}' -c ../../i18next-parser.config.js"
  },
  "jest": {
    "automock": false,
    "setupFiles": [
      "../../setupTests"
    ]
  },
  "bugs": {
    "url": "https://github.com/opensrp/web/issues"
  },
  "dependencies": {
    "@onaio/connected-reducer-registry": "^0.0.3",
    "@onaio/gatekeeper": "^0.1.1",
    "@onaio/session-reducer": "^0.0.12",
    "@onaio/utils": "^0.0.1",
    "@opensrp/notifications": "^0.0.4",
    "@opensrp/pkg-config": "^0.0.4",
    "@opensrp/server-service": "^0.0.16",
    "antd": "^4.6.3"
  },
  "peerDependencies": {
    "@opensrp/store": "^0.0.7",
    "antd": "^4.6.3",
    "react": "^16.18.1",
    "react-router": "^5.2.0",
    "react-router-dom": "^5.2.0"
  },
  "author": "OpenSRP Engineering",
  "license": "Apache-2.0",
  "private": false
}
