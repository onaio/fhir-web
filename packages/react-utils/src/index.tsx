export * from './components/BrokenPage';
export * from './components/Resource404';
export * from './components/UnauthorizedPage';
export * from './components/UtilPageExtra';
export * from './helpers/dataLoaders';
export * from './components/Search';
export * from './components/Search/utils';
export * from './helpers/componentUtils';
export * from './helpers/utils';
export * from './components/LanguageSwitcher';
export * from './components/TableLayout';
