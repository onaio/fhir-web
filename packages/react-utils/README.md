# React Utils

A couple of utils being used in opensrp-web that work with react

## Installation

```sh
yarn add @opensrp/react-utils
```

## Usage

This package currently exposes the following components:

### Broken Page

This is a util component that is shown when a page runs into a non-recoverable error.
it also comes with the `useHandleBrokenPage` hook that helps manage the workflow of a broken page

#### Props/ Configuration

##### title

_Optional_(`string` | defaults `Something went wrong`)

Error title

##### errorMessage

_Optional_(`string` | defaults `Error`)

Error description

##### homeUrl

_Optional_(`string` | defaults `/`)

url path to home page

### Resource404

This is a util component shown when a requested resource is not found;
the more canonical 404 component, is shown when a page is not yet bound
to the routing system, this component is to be used within an existing page
but where one or more resources to be shown on that page are deemed missing.

#### Props/ Configuration

##### title

_Optional_(`string` | defaults `Sorry, the resource you requested for, does not exist`)

Error title

##### errorMessage

_Optional_(`string` | defaults `Error`)

Error description

##### homeUrl

_Optional_(`string` | defaults `/`)

url path to home page

### Code examples

```tsx
import { Resource404, BrokenPage, useHandleBrokenPage } from '@opensrp/react-utils';

const Foo = () => {
  const { broken, handleBrokenPage } = useHandleBrokenPage;
  if (broken) {
    return <BrokenPage />;
  }
  if (!object) {
    return <Resource404 />;
  }
};
```
