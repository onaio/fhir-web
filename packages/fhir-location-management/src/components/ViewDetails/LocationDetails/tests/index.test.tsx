import React from 'react';
import { render, screen } from '@testing-library/react';
import { GeometryRender, LocationDetails } from '../';
import { ContextProvider } from '@opensrp/react-utils';

describe('GeometryRender Component', () => {
  test('renders valid JSON string formatted', () => {
    const validJson = '{"key":"value"}';
    render(<GeometryRender geometry={validJson} />);
    expect(document.querySelector('pre')).toMatchSnapshot();
  });

  test('renders invalid JSON string as is', () => {
    const invalidJson = 'invalid-json';
    render(<GeometryRender geometry={invalidJson} />);
    const preElement = screen.getByText(invalidJson);
    expect(preElement).toBeInTheDocument();
  });

  test('renders empty string when geometry is undefined', () => {
    render(<GeometryRender />);
    const preElement = document.querySelector('pre');
    expect(preElement).toBeInTheDocument();
  });

  test('renders empty string when geometry is an empty string', () => {
    render(<GeometryRender geometry="" />);
    const preElement = document.querySelector('pre');
    expect(preElement).toBeInTheDocument();
  });
});

describe('LocationDetails Component', () => {
  const mockLocation = {
    id: 'loc123',
    name: 'Test Location',
    extension: [
      {
        url: 'http://build.fhir.org/extension-location-boundary-geojson.html',
        valueAttachment: {
          data: 'ewogICAgICAgICJ0eXBlIjogIlBvaW50IiwKICAgICAgICAiY29vcmRpbmF0ZXMiOiBbMC4wLCAwLjBdCiAgICAgIH0=',
        },
      },
    ],
    position: {
      latitude: -14.09989,
      longitude: 36.09809,
      altitude: 1200,
    },
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders with complete location details', () => {
    render(
      <ContextProvider>
        <LocationDetails location={mockLocation} />
      </ContextProvider>
    );

    expect(document.querySelector('body')?.textContent).toMatchInlineSnapshot(
      `"Test LocationEdit detailsID: loc123Version: Date Last UpdatedLocation NameTest LocationStatusaliasLatitude & Longitude-14.09989, 36.09809Physical TypeGeometryZXdvZ0lDQWdJQ0FnSUNKMGVYQmxJam9nSWxCdmFXNTBJaXdLSUNBZ0lDQWdJQ0FpWTI5dmNtUnBibUYwWlhNaU9pQmJNQzR3TENBd0xqQmRDaUFnSUNBZ0lIMD0=Administrative LevelDescription"`
    );

    // Check EditLink rendering
    expect(screen.getByText('Edit details')).toBeInTheDocument();
  });

  test('handles missing fields gracefully', () => {
    const partialLocation = {
      id: 'loc123',
      name: 'Test Location',
    };

    render(
      <ContextProvider>
        <LocationDetails location={partialLocation} />
      </ContextProvider>
    );

    // Check missing fields are not rendered
    expect(screen.queryByText('alias:')).not.toBeInTheDocument();
    expect(screen.queryByText('Latitude & Longitude:')).not.toBeInTheDocument();
    expect(screen.queryByTestId('geometry')).not.toBeInTheDocument();
    expect(screen.queryByText('Administrative Level:')).not.toBeInTheDocument();
    expect(screen.queryByText('Description:')).not.toBeInTheDocument();
  });
});
