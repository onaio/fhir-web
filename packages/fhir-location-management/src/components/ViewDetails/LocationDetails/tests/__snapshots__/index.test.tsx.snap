// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeometryRender Component renders valid JSON string formatted 1`] = `
<pre
  style="max-height: 60px; max-width: 120px; overflow: auto; border: 1px solid #ddd; padding: 4px;"
>
  {
  "key": "value"
}
</pre>
`;
